### DISCUSSION TITLE
gradle Spotless and sA Command's not working after 334 PR merged

### DISCUSSION THREAD

**Discussion by MuhammadSalman-7214** [google/android-fhir#348] [General]
Gradle Spotless and sA Command's not working after 334 PR merged.
![Screenshot from 2021-03-19 14-18-24](https://user-images.githubusercontent.com/64642723/111758199-02f97a00-88be-11eb-9505-971eb6452258.png)

**Comment by epicadk** [comment]
Interesting. I usually just run it in android studio. Can you trying running it that way? Instead of pressing `enter` just press `ctrl`+`enter` after you type `gradlew sA` . I think it's called Smart command execution.

**Reply by MuhammadSalman-7214** [reply]
With ctrl+enter, it shows the same error again, in the following way
![Screenshot from 2021-03-19 15-20-52](https://user-images.githubusercontent.com/64642723/111765965-cbdb9680-88c6-11eb-92f1-6a10ed104036.png)

**Reply by epicadk** [reply]
Do you have node installed? 
I think the plugin requires node as mentioned [here](https://github.com/prettier/plugin-xml) not a 100% sure though.

**Reply by MuhammadSalman-7214** [reply]
not installed yet, let me install it

**Comment by MuhammadSalman-7214** [comment]
by running the following command npm is installed, and after npm installed the problem has been resolved now, no need of installing node, and all the commands are running perfectly fine now.

sudo apt install npm

**Reply by shoaibmushtaq25** [reply]
Thanks @MuhammadSalman-7214 and @epicadk for the guidance here. I was also facing this Let me also install the node. Hope , this would solve the problem for me as well.

**Reply by jingtang10** [reply]
I added https://github.com/google/android-fhir/wiki/Developer's-Guide#nodejs to the wiki

**Reply by MuhammadSalman-7214** [reply]
your welcome @shoaibmushtaq25 :+1: .

**Reply by MuhammadSalman-7214** [reply]
Thanks @jingtang10 for updating the wiki

**Comment by Syed-Affan-Hameed** [comment]
Hi @MuhammadSalman-7214 ,
Recently ,I am finding difficulty in forking this project and running it locally. I have to fix some issues can you help me out brother?.

**Reply by s-ayush2903** [reply]
For setting up, see the instructions from [developers' guide at wiki](https://github.com/google/android-fhir/wiki/Developer's-Guide), in a nutshell, use [canary build version](https://developer.android.com/studio/preview) to build the project and it'll run smooth.

**Reply by s-ayush2903** [reply]
Also, feel free to ask if you encounter any problem(s) : )

**Reply by Syed-Affan-Hameed** [reply]
Hi @s-ayush2903,
Thanks a lot for that.

**Reply by MuhammadSalman-7214** [reply]
if you need any help, you can ask @Syed-Affan-Hameed
